response.go

```diff
@@ -111,17 +111,33 @@ func (r *Response) apiError() error {
 		b.statusCode = r.StatusCode
 		return &b
 	}
-	if r.StatusCode > 299 {
-		return fmt.Errorf("%s (%d)", http.StatusText(r.StatusCode), r.StatusCode)
-	}
 	return nil
 }
 
+func (r *Response) httpError() error {
+	switch r.StatusCode {
+	case 200, 201, 422:
+		return nil
+	default:
+		return httpError(r.StatusCode)
+	}
+}
+
```

Comment:
If you do the changes in the two comments above you should be able to drop this function.